content/open-source/concepts/graphql/graphql.md

```diff
@@ -15,27 +15,72 @@ CatalogContent:
 
 ## Why Adopt GraphQL?
 
-Before adopting GraphQL, developers should be familiar with RESTful architecture.
+Before adopting GraphQL, developers should be familiar with RESTful architecture. Here are some of the benefits of using GraphQL over REST:
 
-- One of REST's disadvantages is it can result in an excessive number of [HTTP](https://www.codecademy.com/resources/docs/general/http) queries or too much data being fetched.
-- Alternatively, GraphQL allows developers to ask for specific data instead of fetching a large object with all of the data every time. There is no room for errors like underfetching and overfetching with GraphQL.
-- Finally, GraphQL allows developers to easily integrate with [databases](https://www.codecademy.com/resources/docs/general/database), [JSON](https://www.codecademy.com/resources/docs/general/json) files, RESTful APIs, and cloud services.
+- GraphQL allows developers to easily integrate with [databases](https://www.codecademy.com/resources/docs/general/database), [JSON](https://www.codecademy.com/resources/docs/general/json) files, RESTful APIs, and [cloud services](https://www.codecademy.com/resources/docs/cloud-computing/cloud-services).
+- It prioritizes giving clients the exact data required and nothing more, so only the needed data is received, helping to eliminate over-fetching.
+- It can be used to construct requests that pull data from multiple sources into a single API call and response.
+- Easier to adopt new data to existing objects, helping to improve future changes and potential under-fetching.
+- For example, an existing setup with objects and types already defined needs to include more data in the object (this could require a large change of including an additional fetch within the flow) or additional steps when modeling/mapping the responses to the object.
 
 ## How GraphQL Works
 
 A common GraphQL workflow can be seen below:
 
 1. First, specific data is requested from the client via a query to GraphQL server.
 2. To retrieve that data, a function on the GraphQL server is invoked.
-3. Lastly, GraphQL server returns the specific data requested by the client.
+3. Lastly, the GraphQL server returns the specific data requested by the client.
+
+In a system like REST, only a single set of arguments can be passed in: the query parameters and URL segments of the request. But in GraphQL, every field and nested object has its own set of arguments, making GraphQL a complete replacement for making multiple API fetches. By passing arguments into scalar fields, data transformations can be implemented on the server rather than on every individual client, which can be more efficient.
```

Comment:
```suggestion
In a system like REST, only a single set of arguments can be passed in: the query parameters and [URL](https://www.codecademy.com/resources/docs/general/url) segments of the request. But in GraphQL, every field and nested object has its own set of arguments, making GraphQL a complete replacement for making multiple API fetches. By passing arguments into scalar fields, data transformations can be implemented on the server rather than on every individual client, which can be more efficient.
```